src/FriendLibraryPage.tsx

```diff
@@ -59,7 +63,7 @@ const FriendLibraryPage: React.FC = () => {
       .finally(() => {
         setIsLoaded(true);
       });
-  }, [id]);
+  }, [id, userStatus]);
```

Comment:
Ah, clever! Does this prevent that infinite fetch loop while still updating the button?

Reply:
Yeah! Exactly. The page needs to refetch the books data only when the user status changes